File: Sample/MaaSAdvertising-iOS-Sample/VideoViewController.m
//
//  VideoViewController.m
//  PWAdvertising
//
//  Created on 4/16/16.
//  Copyright © 2016 Phunware, Inc. All rights reserved.
//

#import "VideoViewController.h"
#import <PWAdvertising/PWAdsVideoAd.h>
#import <PWAdvertising/PWAdsVideoInterstitial.h>
#import <PWAdvertising/PWAdsRequest.h>

// Example zone id, don't use this in your app!
static NSString * const kZoneIDVideo =  @"22219";

@interface VideoViewController () <PWAdsVideoInterstitialDelegate>

@end

@implementation VideoViewController {
    PWAdsVideoInterstitial *_videoInterstitial;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    // Create PWAdsVideoInterstitial instance
    _videoInterstitial = [PWAdsVideoInterstitial new];
    _videoInterstitial.delegate = self;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


- (IBAction)loadButtonPressed:(id)sender {
    // Create ad request instance
    PWAdsRequest *adsRequest = [PWAdsRequest requestWithZoneID:kZoneIDVideo];
    
    // Set test mode for request. This will fetch test ads.
    adsRequest.testMode = YES;
    
    // Load video ads request
    [_videoInterstitial loadAdsRequest:adsRequest];
}

#pragma mark - PWAdsVideoInterstitialDelegate

// Called when the adsLoader receives a video and is ready to play.
- (void)videoInterstitialDidLoadAd:(PWAdsVideoInterstitial *)videoInterstitial {
    NSLog(@"videoInterstitialDidLoadAd:");
}

- (void)videoInterstitialDidFinishedPreCaching:(PWAdsVideoInterstitial *)videoInterstitial {
    NSLog(@"videoInterstitialDidFinishedPreCaching:");
    
    // Video ad is pre-cached. Now present the video interstitial.
    [_videoInterstitial presentFromViewController:self];
}

// Called when a video interstitial fails to load ad.
- (void)videoInterstitial:(PWAdsVideoInterstitial *)videoInterstitial didFailError:(NSError *)error {
    NSLog(@"videoInterstitial:didFailError:");
}

// Called before video interstitial dismisses video modal.
- (void)videoInterstitialWillDismissModal:(PWAdsVideoInterstitial *)videoInterstitial {
    NSLog(@"videoInterstitialWillDismissModal:");
}

// Called after video interstitial dismisses video modal.
- (void)videoInterstitialDidDismissModal:(PWAdsVideoInterstitial *)videoInterstitial {
    NSLog(@"videoInterstitialDidDismissModal:");
}

@end

File: Samples/Sample/Sample/VideoViewController.swift
//
//  VideoViewController.swift
//  Sample
//
//  Created on 8/14/17.
//  Copyright © 2017 Phunware, Inc. All rights reserved.
//

import UIKit
import PWAdvertising

struct VideoInterstitialConstants {
    static let zoneId = "75395"
    static let userId = "1234"
}

class VideoViewController: UIViewController {

    var request = PWAdsRequest(zoneID: VideoInterstitialConstants.zoneId)!
    var videoInterstitial = PWAdsVideoInterstitial()
    
    @IBAction func load(_ sender: UIButton) {
        request.userID = VideoInterstitialConstants.userId
        request.testMode = true
        print("Zone Id: \(request.zoneID!) \nRV User: \(request.userID!) \nTest Mode: \(request.testMode) \n")
        videoInterstitial.load(request)
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        videoInterstitial.delegate = self
    }
}

extension VideoViewController: PWAdsVideoInterstitialDelegate {
    func videoInterstitialDidLoadAd(_ videoInterstitial: PWAdsVideoInterstitial!) {
        print("videoInterstitialDidLoadAd\n")
    }
    
    func videoInterstitial(_ videoInterstitial: PWAdsVideoInterstitial!, didFailError error: Error!) {
        print("rewardedVideo didFailError:\n")
        if let error = error {
            print("\(error)\n")
        }
    }
    
    func videoInterstitialDidDismissModal(_ videoInterstitial: PWAdsVideoInterstitial!) {
        print("videoInterstitialDidDismissModal\n")
    }
    
    func videoInterstitialDidPresentModal(_ videoInterstitial: PWAdsVideoInterstitial!) {
        print("videoInterstitialDidPresentModal")
    }
    
    func videoInterstitialWillDismissModal(_ videoInterstitial: PWAdsVideoInterstitial!) {
        print("videoInterstitialWillDismissModal\n")
    }
    
    func videoInterstitialWillLeaveApplication(_ videoInterstitial: PWAdsVideoInterstitial!) {
        print("videoInterstitialWillLeaveApplication\n")
    }
    
    func videoInterstitialDidFinishedPreCaching(_ videoInterstitial: PWAdsVideoInterstitial!) {
        print("videoInterstitialDidFinishedPreCaching\n")
        videoInterstitial.present(from: self)
    }
}

